components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -433,6 +434,14 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
 
             await mayStartPromise;
 
+            if (runningInstance) {
+                // We already have a running workspace. This may happen if we're forcing the default image.
+                // In that case, we stop the previous first.
+                await this.internalStopWorkspace({ span }, workspaceId, workspace.ownerId).catch(err => {
+                    log.error(logCtx, "stopWorkspace error: ", err);
+                });
```

Comment:
This call only blocks until the stop request is sent.

In order to completely prevent multiple instances from running at the same time, we should probably also listen for instance updates here (e.g. via [messageBusIntegration.listenForWorkspaceInstanceUpdates()](https://github.com/gitpod-io/gitpod/blob/ec9a1922fd09fb195d5c2323c42a48b06169d051/components/server/src/workspace/gitpod-server-impl.ts#L145)) and only move on when the previous instance is definitely stopped.

There is also a case where the previous instance was in a different region, in which case to be fully pedantic we should also poll the DB, but it's probably okay to not do that here.